src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -69,14 +69,25 @@ public static IServiceCollection AddSqlServerConnection(this IServiceCollection
 
             services.AddOptions();
             services.TryAddSingleton<ISqlConnectionStringProvider, DefaultSqlConnectionStringProvider>();
-            services.TryAddSingleton<ISqlConnectionFactory>(
+
+            services.TryAddSingleton<ISqlConnectionBuilder>(
```

Comment:
May I ask to add comment why we try to add two `ISqlConnectionBuilders` here, and how they different?

Reply:
sure